remote_llm.py

```diff
@@ -34,10 +36,15 @@ def generate_text(preprompt, prompt, model):
     
     log.info(f'Received text generation request for prompt "{prompt}" for model {model}')
     # Load the Vanilla Neox API key from the config file.
-
-    raw_response = post(url=route,
-                        headers={'Authorization': f'Bearer {token}'},
-                        json={'prompt': prompt, "preprompt" : preprompt, "model": model})
+    try:
+        raw_response = requests.post(url=route,
+                            headers={'Authorization': f'Bearer {token}'},
+                            json={"uuid": idempotency_uuid,'prompt': prompt, "system" : preprompt, "model": model, "history": history})
+    except requests.Timeout:
```

Comment:
This should work too. Adding logging for extra debugging
```suggestion
    except (requests.Timeout, requests.ConnectionError) as error:
      logging.debug(error)
```